specs/commandline/verify.md

```diff
@@ -86,15 +89,92 @@ cat <<EOF > $HOME/.config/notation/trustpolicy.json
 
 ```shell
 # Prerequisites: Signatures are stored in a registry referencing the signed OCI artifact
 
-# Verify signatures on an OCI artifact identified by the digest
-notation verify registry.wabbit-networks.io/software/net-monitor@sha256:abcd1234abcd1234abcd1234abcd1234abcd1234abcd1234abcd1234abcd1234
+# Verify signatures on an OCI artifact identified by the tag
+$ notation verify localhost:5000/net-monitor:v1
+```
+
+An example of output messages for a successful verification:
+
+```text
+Warning: Tag is used. Always use digest to identify the reference uniquely and immutably.
+
+Resolve tag `v1` to digest `sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9`
+Verify succeeded for localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9
+```
+
+### Verify succeeded using "--debug" flag
+
+```shell
+$ notation verify --debug localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9
+Use trust policy: <policy_name>
+Use certificate: <trust_store_type>/<trust_store_name>/<certificate_name>
+Signature verification level: [strict/permissive/audit/skip]
+
+Verify succeeded on signature ${signature-digest} for localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9
+```
+
+### Verify succeeded with conditions using "--debug" flag
+
+```shell
+$ notation verify --debug localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9
+Use trust policy: <policy_name>
+Use certificate: <trust_store_type>/<trust_store_name>/<certificate_name>
+Signature verification level: [strict/permissive/audit/skip]
+
+Error:
+signature verification failure
+${signature-digest}: ${error message}
+
+Warning:
+signature verification failure, but only logged the error
+${signature-digest}: ${error message}
+
+Verify succeeded on signature ${signature-digest} for localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9
+```
+
+### Verify failed for all signatures without using "--debug" flag
+
+Trust policy and trust store are configured properly.
+
+```shell
+$ notation verify localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9
+Verify failed for all the 2 signature(s) associated with localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9
+```
+
+### Verify failed for all signatures using "--debug" flag
+
+Trust policy and trust store are configured properly.
+
+```shell
+$ notation verify --debug localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9
+Use trust policy: <policy_name>
+Use certificate: <trust_store_type>/<trust_store_name>/<certificate_name>
+Signature verification level: [strict/permissive/audit/skip]
+
+Error:
+signature verification failed
+${signature-digest}: ${error message}
```

Comment:
Printing just a digest is confusing. We should add text explaining that this is the *signature digest* and not some other digest. 

Also, printing the signature would be helpful for debugging purposes. I would suggest changing this (also in other places) to:
```
Error: Signature verification failed for artifact <artifact-registry/artifact-repo:digest>
Detailed error: <if the error provides enough context to be helpful>
Signature digest: <signature-digest>
Signature: <the actual signature>
```

Reply:
Just to clarify the `signature digest` suggested by you is the digest to identify the signature blob, not the digest to identify the signature manifest, right?

For adding the `signature` info, there are two problems:
- There could be many signatures failed, this will lead to very big output message.
- cose signature is in binary format, which will mess the console if the cose signature is printed out.

Reply:
Output message when debug option is enabled will be addressed by other PRs